### ISSUE TITLE
[Lecture 10-2][0612] Meaning of "Weight decay removes the effect of old parameters"

### ISSUE THREAD

**Issue by yewonkim01** [PiLab-CAU/ComputerVision-2401#43]
![image](https://github.com/PiLab-CAU/ComputerVision-2401/assets/115199510/43d5bf1f-9bda-4f0c-8fc5-4f2270671ebc)

I don't quite understand the idea that "weight decay removes the effect of old parameters." I know that weight decay is a regularization technique used to prevent overfitting by reducing large weights, which helps the model generalize better. Could you explain what it means by "weight decay removes the effect of old parameters"? Does it simply mean that weight decay reduces the values of large weights?

**Comment by goals4292** [comment]
I think I can help your understanding with simple equations.

As you said, weight decay is a regularization technique that adds a term to loss function to prevent overfitting.

The typical form of a loss functionL $L(θ)$ with a weight control term added looks like this:

$$
L_{\text{total}}(\theta) = L(\theta) + \lambda \cdot \frac{1}{2} ||\theta||^2 
$$

According to this modified loss function $L_{\text{total}}(\theta)$, the standard weight update equation  $\theta_{\text{new}} = \theta_{\text{old}} - \eta \nabla L(\theta_{\text{old}})$ is modified as

$$
\theta_{\text{new}} = \theta_{\text{old}} - \eta (\nabla L(\theta_{\text{old}}) + \lambda \theta_{\text{old}})
$$



$$
\Rightarrow \theta_{\text{new}} = (1 - \eta\lambda)\theta_{\text{old}} - \eta \nabla L(\theta_{\text{old}})
$$

As you can see the equation above, weight decay multiplies $\theta_{\text{old}}$ by $(1 - \eta\lambda)$ , which results removing the effect of old parameters.

**Comment by yewonkim01** [comment]
Thank you for your response. The formula helped me understand it. Thanks a lot.
